plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java

```diff
@@ -564,13 +572,28 @@ private Composite createButtonsBar( Composite parent ) {
     fileButtons.setLayout( new RowLayout() );
     fileButtons.setLayoutData( new FormDataBuilder().right( 100, 0 ).result() );
 
-    FlatButton
-      upButton =
+
+    flatBtnUp =
       new FlatButton( fileButtons, SWT.NONE ).setEnabledImage( rasterImage( "img/Up_Folder.S_D.svg", 32, 32 ) )
         .setDisabledImage( rasterImage( "img/Up_Folder.S_D_disabled.svg", 32, 32 ) )
         .setToolTipText( BaseMessages.getString( PKG, "file-open-save-plugin.app.up-directory.button" ) )
 
-        .setLayoutData( new RowData() ).setEnabled( true );
+        .setLayoutData( new RowData() ).setEnabled( false ).addListener( new SelectionAdapter() {
+          @Override
+          public void widgetSelected( SelectionEvent selectionEvent ) {
+            TreeSelection treeSelection = (TreeSelection) treeViewer.getSelection();
+            if ( !treeSelection.isEmpty() ) {
+              Object firstElement = treeSelection.getFirstElement();
+              TreePath[] paths = treeSelection.getPaths();
```

Comment:
No issue with logic, just suggestion on coding style. If author don't agree, ignore.

you can move `treeSelection.getPaths();` to inside the second nested if statement, since it is only used there

Reply:
Applied your suggestions, Thank you!